### PR TITLE
add enumeration aggregation projection support

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
## Summary

Fixes https://github.com/finos/legend-studio/issues/1079

https://user-images.githubusercontent.com/29690824/165549910-11135a5a-be51-458f-8dba-f5177317be46.mp4

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-query/src/components/QueryBuilderProjectionPanel.tsx [hunk 1]
```diff
@@ -446,56 +446,83 @@ const QueryBuilderProjectionColumnEditor = observer(
...
-                    tabIndex={-1}
-                    title="Choose Aggregate Operator..."
-                  >
-                    <SigmaIcon />
-                  </button>
-                  <button
-                    className="query-builder__projection__column__aggregate__operator__dropdown__trigger"
-                    tabIndex={-1}
-                    title="Choose Aggregate Operator..."
+                {aggreateOperators.length ? (
```
Comment (akphi) [inline packages/legend-query/src/components/QueryBuilderProjectionPanel.tsx:449]: probably we can use the `disabled` props of `DropdownMenu`

File: packages/legend-query/src/stores/aggregateOperators/QueryBuilderAggregateOperator_JoinString.ts [hunk 2]
```diff
@@ -54,6 +54,7 @@ export class QueryBuilderAggregateOperator_JoinString extends QueryBuilderAggreg
       const propertyType =
         projectionColumnState.propertyExpressionState.propertyExpression.func
           .genericType.value.rawType;
+      //Note: Engine does not support Enumerations at the moment.
```
Comment (akphi) [inline packages/legend-query/src/stores/aggregateOperators/QueryBuilderAggregateOperator_JoinString.ts:57]: Please use the standard form `// NOTE:`

File: .changeset/breezy-experts-complain.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-query': patch
+---
+
+Add enumeration support for aggregation projection columns.
```
Comment (akphi) [inline .changeset/breezy-experts-complain.md:5]: Let's add a separate changeset for the fix of #1079 as we usually do

**Review by akphi — CHANGES_REQUESTED** [review]

File: .changeset/breezy-experts-complain.md (hunk 3 above)
Comment (MauricioUyaguari) [reply .changeset/breezy-experts-complain.md:5]: why a separate one??

File: .changeset/breezy-experts-complain.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-query': patch
+---
+
+Add enumeration support for aggregation projection columns. Fixes ([#1079](https://github.com/finos/legend-studio/issues/1079)).
```
Comment (akphi) [inline .changeset/breezy-experts-complain.md:5]:
```suggestion
Add enumeration support for aggregation projection columns.
Also, aggregation operators dropdown is now disabled when there is no operations compatible with the projection type ([#1079](https://github.com/finos/legend-studio/issues/1079)).
```

File: packages/legend-query/src/stores/aggregateOperators/QueryBuilderAggregateOperator_JoinString.ts [hunk 5]
```diff
@@ -54,6 +54,7 @@ export class QueryBuilderAggregateOperator_JoinString extends QueryBuilderAggreg
       const propertyType =
         projectionColumnState.propertyExpressionState.propertyExpression.func
           .genericType.value.rawType;
+      // Note: Engine does not support Enumerations at the moment.
```
Comment (akphi) [inline packages/legend-query/src/stores/aggregateOperators/QueryBuilderAggregateOperator_JoinString.ts:57]:
```suggestion
      // NOTE: Engine does not support Enumerations at the moment.
```

**Review by akphi — APPROVED** [review]

**Pull request merged**